indexer/src/utils/filter.ts

```diff
@@ -1,71 +1,87 @@
-  }
+/**
+ * Validates if an Ethereum validation has failed based on the event data.
+ *
+ * @param {Event} event - The event containing data to validate.
+ * @returns {boolean} - Returns true if the validation failed, otherwise false.
+ */
+export function ethValidationFailed(event: Event): boolean {
+  // We only need the data array to validate the event
+  const { data } = event;
 
-  if (BigInt(to) !== BigInt(KAKAROT_ADDRESS)) {
-    console.log("✅ Skipping transaction that is not related to Kakarot");
-    return false;
-  }
-  return true;
-};
+  // If data array is empty, return false (no validation failure)
+  if (!data.length) return false;
 
-export const ethValidationFailed = (event: Event) => {
-  const { data } = event;
-  if (data.length === 0) {
-    return false;
-  }
+  // Parse the first element of data as the response length in hexadecimal
+  const responseLen = parseInt(data[0] ?? "", 16);
 
-  const response_len = parseInt(data[0], 16);
+  // Check if the data length is valid (greater than response length + 1)
+  const isValidLength = data.length > responseLen + 1;
 
-  if (response_len + 1 >= data.length) {
+  // If data length is invalid, log an error and return false
+  if (!isValidLength) {
     console.error(
-      `Invalid event data length. Got ${data.length}, expected < ${
-        response_len + 1
+      `Invalid event data length. Got ${data.length}, expected > ${
+        responseLen + 1
       }`,
     );
     return false;
   }
 
-  const success = parseInt(data[1 + response_len], 16);
-  if (success == 1) {
-    return false;
-  }
+  // Parse the element at position (response length + 1) as success flag in hexadecimal
+  const success = parseInt(data[responseLen + 1] ?? "", 16);
 
-  const response = data.slice(1, 1 + response_len);
-  const msg = String.fromCharCode(...response.map((x) => parseInt(x, 16)));
+  // If success flag is set (1), return false (no validation failure)
+  if (success == 1) return false;
 
-  return msg.includes("eth validation failed");
-};
+  // Extract the response data slice, convert it to a string, and check if it includes "eth validation failed"
+  return String.fromCharCode(
+    ...data.slice(1, 1 + responseLen).map((x) => parseInt(x, 16)),
+  ).includes("eth validation failed");
+}
 
-export const isRevertedWithOutOfResources = (receipt: TransactionReceipt) => {
+/**
+ * Checks if a transaction receipt indicates that it was reverted due to running out of resources.
+ *
+ * @param {TransactionReceipt} receipt - The transaction receipt to check.
+ * @returns {boolean} - Returns true if the transaction was reverted due to out of resources, otherwise false.
+ */
+export function isRevertedWithOutOfResources(
+  receipt: TransactionReceipt,
+): boolean {
   return (
     receipt.executionStatus.includes("REVERTED") &&
-    receipt.revertReason?.includes("RunResources has no remaining steps")
+    (receipt.revertReason ?? "").includes("RunResources has no remaining steps")
```

Comment:
I thought the same thing but this syntax throws an error `error: TS2322 [ERROR]: Type 'boolean | undefined' is not assignable to type 'boolean'`. I believe that it is because of the coupling with the `includes` that there is this error, so I preferred this approach which is safer to avoid the error.